src/gwt/acesupport/acemode/markdown_highlight_rules.js

```diff
@@ -125,13 +125,14 @@ var MarkdownHighlightRules = function() {
     };
 
     var linkByUrl = {
```

Comment:
Does this behave well for the 'intermediate' states while the user is typing? E.g.

    [abc](){some text

that is, before the user has inserted the closing `}`.

Reply:
It does, the matching group being entirely optional means nothing breaks with it partially there. Also, since braces are part of another regex, they still get colored when they're not a pair.